cmd/podman/registry/config.go

```diff
@@ -89,7 +89,12 @@ func newPodmanConfig() {
 // use for the containers.conf configuration file.
 func setXdgDirs() error {
 	if !rootless.IsRootless() {
-		return nil
+		// set XDG_RUNTIME_DIR to an empty string for root
+		// Sometimes XDG_RUNTIME_DIR is set to /run/user/0 sometimes it is unset,
+		// the inconsistency is causing issues for the dnsname plugin.
+		// It is already set to an empty string for conmon so lets do the same
+		// for podman. see #10806 and #10745
+		return os.Setenv("XDG_RUNTIME_DIR", "")
```

Comment:
The reason I choose to set it to an empty string is because that's what is done for conmon and the oci runtimes. I agree that unsetting makes sense but I want this to be consistent.

Reply:
Let's unset then we can move to fix the other use cases to also unset.

Reply:
changed to unset